filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala

```diff
@@ -1,23 +1,81 @@
 package cromwell.filesystems.drs
 
+import cats.data.NonEmptyList
 import cloud.nio.impl.drs.DrsCloudNioFileSystemProvider
-import cromwell.core.path.{Path, PathBuilder}
-
-import scala.util.{Failure, Try}
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.core.path.PathFactory.PathBuilders
+import cromwell.core.path.{Path, PathFactory, PreResolvePathBuilder}
 
+import scala.util.{Failure, Success, Try}
 
 case class DrsPathBuilder(fileSystemProvider: DrsCloudNioFileSystemProvider,
-                          requesterPaysProjectIdOption: Option[String]) extends PathBuilder {
+                          requesterPaysProjectIdOption: Option[String],
+                          preResolve: Boolean = false,
+                         ) extends PreResolvePathBuilder with StrictLogging {
 
   private val drsScheme: String = fileSystemProvider.getScheme
 
   override def name: String = "DRS"
 
-  override def build(pathAsString: String): Try[Path] = {
+  override def build(pathAsString: String, pathBuilders: PathBuilders): Try[Path] = {
     if (pathAsString.startsWith(s"$drsScheme://")) {
-      Try(DrsPath(fileSystemProvider.getCloudNioPath(pathAsString), requesterPaysProjectIdOption))
+      Try(createDrsOrOtherPath(pathAsString, pathBuilders))
     } else {
       Failure(new IllegalArgumentException(s"$pathAsString does not have a $drsScheme scheme."))
     }
   }
+
+  private def createDrsOrOtherPath(pathAsString: String, pathBuilders: PathBuilders): Path = {
+    def drsPath = DrsPath(fileSystemProvider.getCloudNioPath(pathAsString), requesterPaysProjectIdOption)
+    if (preResolve) {
+      maybeCreateOtherPath(pathAsString, pathBuilders) getOrElse drsPath
+    } else {
+      drsPath
+    }
+  }
+
+  private def maybeCreateOtherPath(pathAsString: String, pathBuilders: PathBuilders): Option[Path] = {
+
+    def logAttempt[A](description: String, attempt: => A): Option[A] = {
+      logTry(description, Try(attempt))
+    }
+
+    def logTry[A](description: String, tried: Try[A]): Option[A] = {
+      tried match {
+        case Success(result) => Option(result)
+        case Failure(exception) =>
+          logFailure(description, exception)
+      }
+    }
+
+    def logFailure(description: String, reason: Any): None.type = {
+      logger.debug(s"Unable to $description, will use a DrsPath to access '$pathAsString': $reason")
+      None
+    }
+
+    def logSuccess(path: Path): Option[Unit] = {
+      logger.debug(s"Converted '$pathAsString' to '$path'")
+      Option(())
+    }
+
+    val gcsPathOption = for {
+      // If passed an empty list of path builders, returns a None and does not run the conversion
```

Comment:
I don't see a case where this would actually happen. `pathBuilders` comes from `allPathBuilders` which must have contained this `DrsPathBuilder` that we're in right now.

Reply:
I can take this out. It was part of an alternative implementation I started and abandoned where the caller could seed the list of path builders with _only_ the GCS path builder. Since that code isn't there anymore this can go too.

Reply:
removed, thanks!